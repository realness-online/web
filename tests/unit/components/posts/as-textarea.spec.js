import { shallow } from 'vue-test-utils'
import wat from '@/components/posts/as-textarea'
describe ('@/components/posts/as-textarea.vue', () => {
  it ('Renders', () => {
    const wrapper = shallow(wat)
    expect(wrapper.element).toMatchSnapshot()
  })
  it ('#wat_focused() exists', () => {
    expect(typeof wat.methods.wat_focused).toBe('function')
  })
  describe ('#prepare_post', () => {
    it ('Exists', () => {
      expect(typeof wat.methods.prepare_post).toBe('function')
    })
    it ('Only triggers a post event when there is text', () => {
      const wrapper = shallow(wat, {
        data: {
          posts: [],
          new_post: ''
        }
      })
      expect(wrapper.vm.posts.length).toBe(0)
      wrapper.vm.prepare_post()
      expect(wrapper.vm.posts.length).toBe(0)
    })
  })
  describe ('focus', () => {
    it ('Emits a toggle-keyboard event when focused', () => {
      const wrapper = shallow(wat)
      const textarea = wrapper.find('#wat')
      const spy = jest.fn()
      wrapper.vm.$on('toggle-keyboard', spy)
      textarea.trigger('focusin')
      expect(spy).toHaveBeenCalledTimes(1)
    })
    it ('Emits a toggle-keyboard event when loosing focus', () => {
      const wrapper = shallow(wat)
      const textarea = wrapper.find('#wat')
      const spy = jest.fn()
      wrapper.vm.$on('toggle-keyboard', spy)
      textarea.trigger('focusout')
      expect(spy).toHaveBeenCalledTimes(1)
    })
    it ('Emits a post-added event when there is text', () => {
      const wrapper = shallow(wat, {
        data: {
          new_post: 'I like to move it.'
        }
      })
      const textarea = wrapper.find('#wat')
      const spy = jest.fn()
      wrapper.vm.$on('post-added', spy)
      textarea.trigger('focusout')
      expect(spy).toHaveBeenCalledTimes(1)
    })
    it ('Does not emit post-added when there is not text', () => {
      const wrapper = shallow(wat)
      const textarea = wrapper.find('#wat')
      const spy = jest.fn()
      wrapper.vm.$on('post-added', spy)
      textarea.trigger('focusout')
      expect(spy).toHaveBeenCalledTimes(0)
    })
  })
})
