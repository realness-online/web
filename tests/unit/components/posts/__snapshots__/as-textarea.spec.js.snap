// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@/components/posts/as-textarea.vue renders a textarea 1`] = `
<textarea
  cols="1"
  id="wat"
  placeholder=">"
  rows="1"
/>
`;
